### PR TITLE
Implement preprocessing

### PR DESCRIPTION
**Pull request by uellue**
This is useful to allocate data structures for `dtype=object` buffers.

Crystallizing PR #458

### CODE REVIEW

File: docs/source/udf/advanced.rst [hunk 1]
```diff
@@ -147,6 +156,8 @@ indications for it. Implementing
 :meth:`~libertem.udf.UDFPartitionMixin.process_partition` activates
 per-partition processing for an UDF.
 
+.. versionadded:: 0.3.0.dev0
```
Comment (sk1p) [inline docs/source/udf/advanced.rst]: Should be moved to previous paragraph?

Comment (uellue) [reply docs/source/udf/advanced.rst]: Yes!

**Comment by uellue** [comment]
Moved to new changelog as well.

**Comment by sk1p** [comment]
Looks good, thanks!

**Pull request merged**